core-processor/src/processor.rs

```diff
@@ -99,18 +99,18 @@ fn process_error(
         });
     }
 
-    if let Some(message) = generate_trap_reply(&message, initial_nonce) {
+    if !dispatch.is_reply() || dispatch.exit_code().expect("Checked before") == 0 {
```

Comment:
And this is the second place I'm worrying about. Can we eliminate this `expect`? It looks like this function is called only to process failed executions - so seems like `exit_code` cannot be equal to 0. Am I wrong?

Reply:
This is the exit code of incoming dispatch - which was processed. So it may have any exit code. Processing error related to the result of execution, not to initial message

Reply:
But I can replace this equality with `dispatch.exit_code() == Some(0)` if you really want.